hydra-test-utils/src/Test/Hydra/Prelude.hs

```diff
@@ -61,8 +62,20 @@ withTempDir :: String -> (FilePath -> IO r) -> IO r
 withTempDir baseName action = do
   tmpDir <- createSystemTempDirectory baseName
   res <- action tmpDir
-  removePathForcibly tmpDir
+  cleanup 0 tmpDir
   pure res
+ where
+  -- NOTE: Somehow, since 1.35.0, cleaning-up cardano-node database directory
+  -- _sometimes_ generates an empty 'clean' file which prevents the 'db' folder
+  -- to be fully removed and triggers an 'UnsatisfiedConstraints' IOException.
```

Comment:
This is an odd workaround and refers to the `cardano-node` behavior, while this function is very generic and lives in `hydra-test-utils`. Can't we do some node-specific cleanup prior to this in the `action`?

Reply:
I know. If you feel like debugging this, go ahead. Spent already an hour chasing this thing and gave up.